tasks/devkit.rake

```diff
@@ -15,17 +16,17 @@ ENV['PATH'] = TDK::OS.join_env_path(
 
 PLAN_FILE = 'plan.tfplan'.freeze
 
-def destroy_if_fails(env)
+def destroy_if_fails(env, task)
   yield
-rescue StandardError => e
+rescue Exception => e
   puts "ERROR: #{e.message}"
   puts e.backtrace.join("\n")
-  task('destroy').invoke(env.name) if env.local_backend?
+  Rake::Task[task_in_current_namespace('destroy', task)].invoke(env.name) if env.local_backend?
   raise
 end
 
 def invoke_if_defined(task_name, env)
```

Comment:
It seems we always call this method using `task_in_current_namespace`. So, it might be simpler to just pass the `task_name` and the `task`, and let `invoke_if_defined` call `task_in_current_namespace` itself.

Reply:
Actually, considering the other usage for `task_in_current_namespace` is in expressions such as:

`Rake::Task[task_in_current_namespace('destroy', task)].invoke(env.name)`

It might be better to create another method called `invoke` which also calls `task_in_current_namespace` itself. Then, the previous code would look much simpler:

`invoke('destroy', task, env.name)`